indico/modules/events/contributions/lists.py

```diff
@@ -66,6 +68,20 @@ def _build_query(self):
                          db.undefer('attachment_count'),
                          db.undefer('is_scheduled')))
 
+    def _build_registration_query(self, is_speaker=False):
+        registration_join_criteria = [
+            Registration.event_id == Contribution.event_id,
+            Registration.is_active,
+            (Registration.user_id == EventPerson.user_id) | (Registration.email == EventPerson.email)
+        ]
+        if is_speaker:
+            registration_join_criteria.append(ContributionPersonLink.is_speaker)
+        return (db.session.query(Contribution.id)
+                .filter(Contribution.event_id == self.event.id)
+                .join(ContributionPersonLink)
+                .join(EventPerson)
+                .join(Registration, db.and_(*registration_join_criteria)))
```

Comment:
I'd keep the `{id_ for id_, in query}` logic for the return value: there's much less obscurity if we return just plain IDs instead of a query object (I couldn't tell you w/o looking at the query log whether SA turns it into a subquery somehow or executes the query when you pass it to .`in_()`...)

Reply:
OK I just checked the log and it generates subqueries within the `IN (...)`. Since it works fine and doesn't cause any problems w/ duplicate references (since there's no correlation to the outer query), not any performance problems, I think we can keep it as it is